mobile_rewards/src/server.rs

```diff
@@ -283,23 +293,39 @@ impl Server {
 
     async fn handle_rewards(
         &mut self,
-        reward_period: RewardPeriod,
-        last_reward_time: i64,
-        end_utc: i64,
+        reward_period: Range<u64>,
+        start_utc: DateTime<Utc>,
+        end_utc: DateTime<Utc>,
     ) -> Result {
         tracing::info!("triggering rewards emissions");
 
-        let store = FileStore::from_env().await?;
-        let rewards = SubnetworkRewards::from_last_reward_end_time(
-            store,
-            &mut self.follower_service,
-            last_reward_time,
-        )
-        .await?;
+        let file_list = self
+            .verifier_store
+            .list_all(FileType::OwnerEmissions, start_utc, end_utc)
+            .await?;
+
+        // TODO: We need a better pattern for this:
+        let mut stream = self
+            .verifier_store
+            .source(stream::iter(file_list).map(Ok).boxed());
+        let mut rewards = vec![];
+        while let Some(Ok(msg)) = stream.next().await {
+            match OwnerEmissions::decode(msg) {
+                Ok(OwnerEmissions { emissions }) => {
+                    for emission in emissions.into_iter() {
+                        rewards.push(SubnetworkReward {
+                            account: emission.pub_key,
+                            amount: emission.weight,
```

Comment:
is this the correct value that should be pushed into the reward? either the name `weight` is misleading or this is the weighted share of a given amount of rewards, not the actual rewards themselves. i believe this needs to have maths done on it to determine how many mobile bones of the total for the reward period should be allocated to each owner pubkey based on the weight amount. the mobile bones integer is what should be setting the reward amount value

Reply:
I think I need to remove owner emissions and just use subnetwork rewards.  I probably just named the field incorrectly 

```rust
impl From<OwnerEmissions> for SubnetworkRewards {
    fn from(owner_emissions: OwnerEmissions) -> SubnetworkRewards {
        let unsorted_rewards = owner_emissions
            .0
            .into_iter()
            .map(|(owner, amt)| ProtoSubnetworkReward {
                account: owner.to_vec(),
                amount: u64::from(amt),
            })
            .collect();
        SubnetworkRewards(unsorted_rewards)
    }
}
```

Reply:
https://github.com/helium/proto/pull/178